vscode/oso/client/src/extension.ts

```diff
@@ -154,6 +159,9 @@ async function startClient(folder: WorkspaceFolder, context: ExtensionContext) {
   context.subscriptions.push(deleteWatcher);
   context.subscriptions.push(createChangeWatcher);
 
+  // Synchronize `events` state across devices.
+  context.globalState.setKeysForSync([telemetryEventsKey]);
```

Comment:
Are there any docs you were looking at for this synchronization API?

I'm not sure this is the intended usage, because it [looks like](https://code.visualstudio.com/api/extension-capabilities/common-capabilities#setkeysforsync-example) it's designed for settings sync (infrequently changing data, rather than high frequency like telemetry).

It also will only be enabled if the user logs into VSCode & turns on settings sync. Not sure whether that's an issue for us.

Reply:
> Are there any docs you were looking at for this synchronization API?

The one you linked, plus docs linked off of it like [this](https://code.visualstudio.com/docs/editor/settings-sync#_sync-user-global-state-between-machines) and looking through Microsoft's org for [examples](https://github.com/search?p=1&q=org%3Amicrosoft+setKeysForSync&type=Code).

> I'm not sure this is the intended usage, because it looks like it's designed for settings sync (infrequently changing data, rather than high frequency like telemetry).
>
> It also will only be enabled if the user logs into VSCode & turns on settings sync. Not sure whether that's an issue for us.

Might not be designed for this use case; I'm not sure. I haven't noticed any performance impact from using it. Most of this telemetry stuff is happening asynchronously in the background so not super worried about it.

Reply:
Removed in #1414